android-sdk/README.md

```diff
@@ -0,0 +1,186 @@
+    private static final Map<String, String> dependencyMap = new HashMap<>();
+    /**
+     * Register the looking for a service module to a class.
+     *
+     * @param type the type of module to use
+     * @param moduleClass the class that implements a service module type
+     * @param dependsOn the other serviceTypes this Service Depends on.
+     */
+    public static void registerServiceModule(String type, Class<? extends ServiceModule> moduleClass, String... dependsOn) {
+        serviceTypeMap.put(type, moduleClass);
+        for (String dependency : dependsOn) {
+            dependencyMap.put(type, dependency);
+        }
+    }
+
+}
+```
+** Discussion Point ** I'm not a fan of static methods during testing.  Perhaps there should be a method in the ServiceModule interface which returns a list of services that the service depends on.  MobileCore could then read that list by invoking that method as part of bootstrapping.
+
+#### Referencing a Service
+An instance of the core bassed during bootstrap can be used to reference other services.  MobileCore is responsible for ensuring that ServiceModule.bootstrap is not called until all declared dependencies are bootstrapped.
+
+SyncService.java
+```
+public class SyncService implements ServiceModule {
+
+  KeyCloakService keycloakService;
+  /*snip*/
+  public void bootstrap(MobileCore core, ServiceConfiguration config) {
+     this.keycloakService = core.getService("keycloak", KeyCloakService.class);
+     this.networkService = core.getService("network", NetworkService.class);
+     /*snip*/
+  }
+
+  public void performSync() {
+    if (keycloakService.isLoggedIn()) {
+      Reuqest request = networkService.newRequest();
+      request.addSecurityHeaders(keycloakService.getHeaders());//pseudocode method, don't expect this exactly.
+      request.setData(this.buildrequestBody());
+      Response response = networkService.execute(request);
+      response.onError(/*Error handling lambda*/);
+      response.onSuccess(/*Success handling lambda*/);
+    }
+  }
+
+}
+```
+
+#### Using Core Threads
+
+The Mobile Core SDK will provide Java Executors that wrap several common blocking tasks.  These executors will be available to any module or part of the application which may need to schedule diskIO work, netowrking work, or work to happen on the main thread.
+
+```
+AppExecutors.networkThread.execute(() -> {
+  var result = someRequest.call(); //Perform a HTTP operation off the main thread
+  AppExecutors.mainThread.execute(()->{updateDisplay(result);});//Update the Android UI on the main thread.
+ });
+```
+
+#### Using Core Networking
+For now, networking will be a service exposed by a http service module.  The definition of this service will be a different proposal.
+
+#### Proposal diagram
+![Core Service and Module diagram](./img/diagram.png)
+
+
+### Misc. Bits
+The following topics are not required 
```

Comment:
I'm not understanding this line in this context.
Do you mean you don't think any of these should be done?
Or they should be done, but not part of this proposal?

Reply:
Based on the primary goal, the sub sections here should be given more importance than 'not required'

* as well as define for Android what tools we will build (ie build plugins, IDE tooling, scripts, etc)
* release and distribution processes
* and processes for ensuring code quality (dependency management, static analysis, CI/CD, etc)

If they aren't required, should they be part of the primary goal then?

Reply:
They should be done, but might not be part of this proposal.

The proposal has narrowed its scope in the past week or so from a "all powerful android sdk proposal" to "Core SDK setup and module system" which is much more manageable.  This language may need to be reworked in that light.

Reply:
OK. It may be enough to remove them from the Primary Goals (or move them to a Non Goals section if you still want to draw some attention to them).